dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/tools/ParallelDependentTestFinder.java

```diff
@@ -0,0 +1,600 @@
+ *         attain the same specified result in a new test execution order.
+ */
+package edu.washington.cs.dt.impact.tools;
+
+import edu.washington.cs.dt.RESULT;
+import edu.washington.cs.dt.TestExecResult;
+import edu.washington.cs.dt.impact.util.Constants;
+import edu.washington.cs.dt.runners.FixedOrderRunner;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Usages:
+ * Get ALL_DT_LIST for just one test:
+ * <pre>
+ * {@code
+ * DependentTestFinder dtFinder = new DependentTestFinder(dependentTestName, origOrder, newOrder, filesToDelete);
+ * List<String> allDtList = dtFinder.runDTF().dependencyToString();
+ *
+ * // Optional write to file:
+ * DependentTestFinder.writeToFile(dtFinder, FILE);
+ * DependentTestFinder.writeToFile(dtFinder); // Writes to file stored in static field DT_FILE
+ * }
+ * </pre>
+ *
+ * Get ALL_DT_LIST for multiple tests, using same orig/new order:
+ * <pre>
+ * {@code
+ * List<DependentTestFinder> dtFinders = new ArrayList<>();
+ *
+ * For first test:
+ * DependentTestFinder dtFinder = new DependentTestFinder(dependentTestName, origOrder, newOrder, filesToDelete);
+ *
+ * for each extra dt to fix:
+ * for (dependentTestName : dependentTestsToFix) {
+ *     dtFinders.add(dtFinder.runDTF());
+ *     dtFinder = dtFinder.createFromDTFinder(dtFinder, nextDependentTestName);
+ * }
+ *
+ * When done:
+ *
+ * List<String> allDtList = DependentTestFinder.dependenciesToString(dtFinders);
+ *
+ * // Optional write to file:
+ * DependentTestFinder.writeToFile(dtFinders, FILE);
+ * DependentTestFinder.writeToFile(dtFinders); // Writes to file stored in static field DT_FILE
+ *
+ * </pre>
+ */
+public class ParallelDependentTestFinder {
+    // TODO: Set up way to set this via command line
+    private static File DT_FILE;
+
+    private class TestOrder {
+        private final List<String> testOrder;
+        private final Map<String, RESULT> results;
+
+        public TestOrder(final List<String> testOrder, final Map<String, RESULT> results) {
+            this.testOrder = testOrder;
+            this.results = results;
+        }
+
+        public TestOrder(final List<String> testOrder) {
+            this.testOrder = testOrder;
+
+            results = runTestOrder(testOrder).getNameToResultsMap();
+        }
+
+        public List<String> getTestsBefore(final String testName) {
+            return testOrder.subList(0, testOrder.indexOf(testName));
+        }
+    }
+
+    private static TestExecResult runTestOrder(final List<String> order) {
+        return new FixedOrderRunner(order).run().getExecutionRecords().get(0);
+    }
+
+    public static void main(String[] args) {
```

Comment:
You can remove this main method. We have no intention of letting the class be used this way at least for now.